ignite/contrib/handlers/visdom_plotter.py

```diff
@@ -0,0 +1,212 @@
+from ignite.engine import Engine
+from ignite.engine import Events
+import numpy as np
+from typing import Callable, List
+
+import visdom
+
+
+class VisdomPlotter:
+    """Handler that plots metrics using Visdom graphs.
+
+    The `VisdomPlotter` can be used to plot to multiple windows each one with a different
+    set of metrics.
+
+    Args:
+        vis (Visdom object, optional): Visdom client.
+        server (str, optinal): URL of visdom server.
+        env (str, optional): Name of Visdom environment for the graphs. Defaults to "main".
+        log_to_filename (str, optional): If given, the plots will be also be save to a file
+            by this name. Later this graphs can be replayed from this file.
+        save_by_default (bool, optional): The graphs will be saved by default by the server.
+
+    Note:
+        Either the `vis` or `server` arguments should be given to the constructor.
+
+    Examples:
+
+    Plotting of trainer loss.
+
+    .. code-block:: python
+
+        trainer = create_supervised_trainer(model, optimizer, loss)
+
+        visdom_plotter = VisdomPlotter(vis, env=env)
```

Comment:
Visdom initialization can fail the whole run in the middle of the code. Maybe we could initialize the connection to the server in the file itself "ignite/contrib/handlers/visdom_plotter.py" (and ensure that connection is established on import of visdom_handler)...

Reply:
In most situations, visdom is supposed to run on a separate server. Therefore trying to establish connection on import of the module will fail as there is a need for the server url. The constructor of the handler has the option to pass the server url and establish the connection.

Reply:
Yes, you are right, I forgot about this. Maybe we can define them as environment variables ?

Reply:
@amitibo coming back to this one again. IMO user is not intended to reconfigure `VisdomPlotter` on multiple servers. There is a single server running in background and user should know a priori the url. We can thus require to define it as `VISDOM_SERVER_URL` environment variable and check if everything is connectable during import and avoid problems in the middle the code execution. What do you think ?